Create draft release notes for 2.4.0

gassmoeller opened the pull request:
A draft for release notes for 2.4.0. Related to #4841.

@geodynamics/aspect-developers: any suggestions for changes? I tried to take the most important changes from https://aspect.geodynamics.org/doc/doxygen/changes_current.html, but that is of course subjective, happy to adjust if I missed something important.

jdannberg commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+  (Wolfgang Bangerth)
+
+- New: There is now a new material model for melting in the lowermost mantle.
+  It can be used to reproduce the results of Dannberg, et al. (2021).
```
```suggestion
  It can be used to reproduce the results of Dannberg et al. (2021).
```

jdannberg commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+  It can be used to reproduce the results of Dannberg, et al. (2021).
+  (Juliane Dannberg)
+
+- New: The geoid postprocessor can now handle a deforming mesh, next to the
```
```suggestion
- New: The geoid postprocessor can now handle a deforming mesh, in addition to the
```

jdannberg commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+  sediment and rock material infilling the flexural moat.
+  (Daniel Douglas)
+
+- New: ASPECT now has a cookbook which uses the gravity postprocessor to
```
```suggestion
- New: ASPECT now has a cookbook that uses the gravity postprocessor to
```

jdannberg commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+  compute gravity generated by S40RTS-based mantle density variations.
+  (Cedric Thieulot)
+
+- New: ASPECT now has a cookbook which shows how velocities can be prescribed
```
```suggestion
- New: ASPECT now has a cookbook that shows how velocities can be prescribed
```

jdannberg reviewed (approved):
This list looks good to me. Would it be worth mentioning that we removed the reference viscosity? Users would have to make the corresponding changes in their own plugins, so it would be good to make them aware of that.

tjhei commented:
Good. I would like a shorter version with 4-5 points as well.

anne-glerum commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+
+- New: Where possible, when using large data tables as input (e.g., for initial
+  conditions specified as tables), these data are now stored only once on each
+  node in memory areas that is accessible by all MPI processes on that node.
```
```suggestion
  node in memory areas that are accessible by all MPI processes on that node.
```

tjhei replied:
This is still open.

anne-glerum commented on contrib/release/release-tasklist:
```diff
@@ -130,6 +130,129 @@
+  (Juliane Dannberg, Rene Gassmoeller)
+
+- New: Added a new postprocessor which computes the parameter "Mobility"
+  following Lourenco et al., 2020 G3
```
```suggestion
  following Lourenco et al., 2020 G3.
```

anne-glerum reviewed (commented):
Do the fixes for compressible models (item 31 and 31 of https://aspect.geodynamics.org/doc/doxygen/changes_current.html) require their own item, or do they fall under the item 'bug fixes'?

gassmoeller commented:
I addressed the comments.

For a shorter version, I would suggest:
```
- New: The matrix-free GMG Stokes solver now works for problems with
  free-surface boundaries, elasticity, and the Newton solver.
  (Jiaqi Zhang, Timo Heister, Anne Glerum, Menno Fraters, John Naliboff)

- New: Where possible, when using large data tables as input (e.g., for initial
  conditions specified as tables), these data are now stored only once on each
  node in memory areas that are accessible by all MPI processes on that node.
  (Wolfgang Bangerth)

- New: There is now a new material model for melting in the lowermost mantle.
  It can be used to reproduce the results of Dannberg et al. (2021).
  (Juliane Dannberg)

- New: The least squares particle interpolation plugins now provide a bound
  preserving slope limiter that respects local bounds on each cell. Particle operations
  have been significantly optimized, in particular in combination with a recent 
  deal.II version (9.4.0 or newer).
  (Mack Gregory, Gerry Puckett, Rene Gassmoeller)

- There are new cookbooks and benchmarks for load induced flexure, the 
  global gravity field, prescribed velocities, kinematically driven oceanic
  subduction, visualizing phase diagrams, and phase transitions in mantle
  convection models.
  (Daniel Douglas, Cedric Thieulot, Bob Myhill, Anne Glerum,
  Haoyuan Li, Magali Billen, Juliane Dannberg)

- Fixed: Many bugs, see link below for a complete list.
  (Many authors. Thank you!).

A complete list of all changes and their authors can be found at
  https://aspect.geodynamics.org/doc/doxygen/changes_between_2_83_80_and_2_84_80.html

Wolfgang Bangerth, Juliane Dannberg, Menno Fraters, Rene Gassmoeller,
Anne Glerum, Timo Heister, Bob Myhill, John Naliboff,
and many other contributors.
```

Agreed?

tjhei commented:
Can you rebase please? I want to see if the tester succeeds. :-)

gassmoeller commented:
Rebased.

tjhei commented:
You picked up the deprecation warning that should have been fixed by #4918
